### PR TITLE
[JENKINS-64520][JENKINS-62340] Pass ssh host key to ssh process when using the strategy CHECK_NEW_HARD

### PR DESCRIPTION
**Pull request by nigelarmstrong-bc**
Pass ssh host key to ssh process when using the strategy CHECK_NEW_HARD. Without this when connecting via ssh process, the ssh process might not have the ssh host key. Take the ssh host key that jenkins already verified and pass that to the ssh process. 

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java [hunk 1]
```diff
@@ -322,6 +333,36 @@ private File createIdentityKeyFile(EC2Computer computer) throws IOException {
...
+        HostKey ec2HostKey = HostKeyHelper.getInstance().getHostKey(computer);
+        if (ec2HostKey == null){
+            return null;
+        }
+        File tempFile = File.createTempFile("ec2_", "_known_hosts");
+        String knownHost = "";
+        knownHost = String.format("%s %s %s", ec2HostAddress, ec2HostKey.getAlgorithm(), Base64.getEncoder().encodeToString(ec2HostKey.getKey()));
+
+        try {
+            FileOutputStream fileOutputStream = new FileOutputStream(tempFile);
```
Comment (res0nance) [inline src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java:346]: This bit will probably just read better in a try with resources way
try (OutputStreamWriter writer = new OutputStreamWriter(new FileOutputStream(tempFile), StandardCharsets.UTF_8))

which will take care of closing it on an exception

**Review by res0nance — APPROVED** [review]
Looks ok untested

File: src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java (hunk 1 above)
Comment (nigelarmstrong-bc) [reply src/main/java/hudson/plugins/ec2/ssh/EC2UnixLauncher.java]: I've given this a shot. My java isn't great, so would appreciate a double check of my changes.

**Comment by nigelarmstrong-bc** [comment]
I tested this on our installation with `CHECK_NEW_HARD` and `CHECK_NEW_SOFT`, but I think it would be good to get more testing on other configurations.

**Review by res0nance — APPROVED** [review]

**Pull request merged**